src/damage.cpp

```diff
@@ -190,7 +190,11 @@ void ammo_effects( const tripoint &p, const std::set<std::string> &effects )
             g->m.add_field( pt, fd_tear_gas, 3, 0 );
         }
     }
-
+    if( effects.count( "GAS_FUNGICIDAL" ) > 0 ) {
+        for( auto &&pt : g->m.points_in_radius( p, 1, 0 ) ) {
```

Comment:
I thought you are just asking whether this is the solution to the problem of not having the point of impact.
Because it is: `p` is the last point the projectile reached and so the point of impact and the last possible point of impact at that.

This part is good.